src/go-slang/ece.ts

```diff
@@ -57,10 +69,26 @@ const CALL_MAIN: CallExpression = {
   args: []
 }
 
-export function evaluate(program: SourceFile, context: Context): Value {
+export function evaluate(program: SourceFile, slangContext: SlangContext): Value {
   const C = new Stack<Instruction>()
   const S = new Stack<any>()
-  let E = createGlobalEnvironment()
+  const E = createGlobalEnvironment()
+
+  // inject predeclared functions into the global environment
+  const B = new Map<number, (...args: any[]) => any>()
+  PREDECLARED_FUNCTIONS.forEach(({ name, func, op }, id) => {
+    E.declare(name, { ...op, id })
+    if (name === 'println') {
+      // println is special case where we need to the `rawDisplay` slang builtin for
+      // console capture, therefore we handle it differently from other predeclared functions
+      // NOTE: we assume that the `rawDisplay` builtin always exists
+      B.set(id, func(slangContext.nativeStorage.builtins.get('slangRawDisplay')))
+      return
```

Comment:
A bit unintuitive - would assume builtins would lie directly on slangContext. But understandable. No changes necessary.

Reply:
yea i agree, the slang's `Context` object is incredibly bloated